crawler/src/main/java/com/brandwatch/ahmedanalytics/crawler/CrawlerJob.java

```diff
@@ -14,24 +17,34 @@
 
 @Component
 public class CrawlerJob {
+    private static final ObjectMapper mapper = new ObjectMapper();
 
     private final Twitter twitter;
-    private MentionService mentionService;
+    private final Producer<String, String> mentionProducer;
+
     private QueryService queryService;
 
-    public CrawlerJob(Twitter twitter, MentionService mentionService, QueryService queryService) {
+    public CrawlerJob(Twitter twitter, Producer<String, String> producer, QueryService queryService) {
         this.twitter = twitter;
-        this.mentionService = mentionService;
+        this.mentionProducer = producer;
         this.queryService = queryService;
     }
 
     @Scheduled(fixedRate = 60000)
     public void crawl() {
         List<Query> queries = queryService.findAll();
-
         for (Query query : queries) {
             List<Tweet> tweets = twitter.searchOperations().search(query.getSearchString()).getTweets();
-            mentionService.saveMentions(tweetsToMentions(tweets, query.getId()));
+            List<Mention> mentions = tweetsToMentions(tweets, query.getId());
+
+            mentions.forEach(mention -> {
+                try {
+                    mentionProducer.send(new ProducerRecord<>("mention",
+                            mapper.writeValueAsString(mention)));
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
```

Comment:
Did you investigate logging options? It's better to use logger and add a meaningful message for the error.

Reply:
Logger added 👍